src/libcore/slice.rs

```diff
@@ -666,6 +666,60 @@ macro_rules! iterator {
                 let exact = diff / (if size == 0 {1} else {size});
                 (exact, Some(exact))
             }
+
+            #[inline]
+            fn count(self) -> usize {
```

Comment:
Unfortunately this does change the semantics of `count` from the default implementation which exhausts the iterator. Perhaps this could modify start to equal end so retain equivalent semantics?